platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -58,100 +54,65 @@
      *
      * @param configurationDirectoryPath directory to watch for changes
      * @param fileExtension              extension of of the files to watch
-     * @param watchService               watch service to use
-     * @param executorService            executor service used to create the watch thread
      */
     public ConfigurationFilesPoller(@NotNull Path configurationDirectoryPath,
-            @NotNull String fileExtension, @NotNull WatchService watchService,
-            @NotNull ExecutorService executorService) {
+            @NotNull String fileExtension) {
         notNull(configurationDirectoryPath, "configurationDirectoryPath cannot be null");
         notNull(fileExtension, "fileExtension cannot be null");
-        notNull(watchService, "watchService cannot be null");
-        notNull(executorService, "executorService cannot be null");
 
         this.configurationDirectoryPath = configurationDirectoryPath;
-        this.watchService = watchService;
-        this.executorService = executorService;
         this.fileExtension = fileExtension;
     }
 
-    public void init() throws IOException {
-        LOGGER.debug("Starting {}...",
-                this.getClass()
-                        .getName());
-
+    public void register(@NotNull ChangeListener listener) {
+        notNull(listener, "ChangeListener cannot be null");
+        changeListener = listener;
+        fileAlterationObserver =
+                new FileAlterationObserver(configurationDirectoryPath.toAbsolutePath()
+                        .toString(), new SuffixFileFilter(fileExtension));
+        fileAlterationObserver.addListener(this);
+        watchService = new FileAlterationMonitor(INTERVAL, fileAlterationObserver);
         try {
-            LOGGER.debug("Registering path [{}] with Watch Service.",
-                    configurationDirectoryPath.toString());
-            configurationDirectoryPath.register(watchService, ENTRY_CREATE);
-        } catch (IOException e) {
-            LOGGER.error("Unable to register path [{}] with Watch Service",
-                    configurationDirectoryPath.toString(),
-                    e);
-            throw e;
+            watchService.start();
+        } catch (Exception e) {
+            String msg = "Failed to start, 'Platform :: Migration' must be restarted: ";
+            LOGGER.info(msg, e);
```

Comment:
❓If we really want the stack trace at a different log level, should we use `DEBUG` instead of `INFO`?